modules/gapi/src/backends/fluid/gfluidimgproc_simd_avx2.hpp

```diff
@@ -0,0 +1,191 @@
+
+                v_gather_pairs(src1[line], &mapsx[x], low2, high2);
+                v_deinterleave(low2, high2, s10, s11);
+
+                //  v_float32 res1 = s10*alpha0 + s11*alpha1;
+                v_float32x8 res1 = v_fma(s10 - s11, alpha0, s11);
+                //  v_float32 d = res0*beta0 + res1*beta1;
+                v_float32x8 d = v_fma(res0 - res1, v_beta0, res1);
+
+                v_store(&dst[line][x], d);
+            }
+
+            for (; x < outSz.width; ++x)
+            {
+                float alpha0 = alpha[x];
+                float alpha1 = 1 - alpha0;
+                int   sx0 = mapsx[x];
+                int   sx1 = sx0 + 1;
+                float res0 = src0[line][sx0] * alpha0 + src0[line][sx1] * alpha1;
+                float res1 = src1[line][sx0] * alpha0 + src1[line][sx1] * alpha1;
+                dst[line][x] = beta0 * res0 + beta1 * res1;
+            }
+        }
+    }
+    else if (!xRatioEq1)
+    {
+        GAPI_DbgAssert(yRatioEq1);
+
+        for (int line = 0; line < lpi; ++line) {
+            int x = 0;
+
+            v_float32x8 low, high, s00, s01;
+            for (; x <= outSz.width - nlanes; x += nlanes)
+            {
+                v_float32x8 alpha0 = v256_load(&alpha[x]);
+                //  v_float32 alpha1 = 1.f - alpha0;
+
+                v_gather_pairs(src0[line], &mapsx[x], low, high);
+                v_deinterleave(low, high, s00, s01);
+
+                //  v_float32 d = s00*alpha0 + s01*alpha1;
+                v_float32x8 d = v_fma(s00 - s01, alpha0, s01);
+
+                v_store(&dst[line][x], d);
+            }
+
+            for (; x < outSz.width; ++x) {
+                float alpha0 = alpha[x];
+                float alpha1 = 1 - alpha0;
+                int   sx0 = mapsx[x];
+                int   sx1 = sx0 + 1;
+                dst[line][x] = src0[line][sx0] * alpha0 + src0[line][sx1] * alpha1;
+            }
+        }
+
+    }
+    else if (!yRatioEq1)
+    {
+        GAPI_DbgAssert(xRatioEq1);
+        int length = inSz.width;  // == outSz.width
+
+        for (int line = 0; line < lpi; ++line) {
+            float beta0 = beta[line];
+            float beta1 = 1 - beta0;
+            v_float32x8 v_beta0 = v256_setall_f32(beta0);
+            int x = 0;
+
+            for (; x <= length - nlanes; x += nlanes)
+            {
+                v_float32x8 s0 = v256_load(&src0[line][x]);
+                v_float32x8 s1 = v256_load(&src1[line][x]);
+
+                //  v_float32 d = s0*beta0 + s1*beta1;
+                v_float32x8 d = v_fma(s0 - s1, v_beta0, s1);
+
+                v_store(&dst[line][x], d);
+            }
+
+            for (; x < length; ++x) {
+                dst[line][x] = beta0 * src0[line][x] + beta1 * src1[line][x];
+            }
+        }
+
+    }
+    else
+    {
+        GAPI_DbgAssert(xRatioEq1 && yRatioEq1);
+        int length = inSz.width;  // == outSz.width
+        for (int line = 0; line < lpi; ++line)
```

Comment:
Still memory allocated continuously (i believe) is it possible to make copy whole matrix block ( which is linear data in memory) using single one `memcpy` call instead of wasting CPU cycles on loop iteration?

Reply:
Applied.